src/bed.h

```diff
@@ -23,12 +23,12 @@ class BedItem final : public Item
 	void readAttr(AttrTypes_t attr, OTB::iterator& first, const OTB::iterator& last) override;
 	void serializeAttr(PropWriteStream& propWriteStream) const override;
 
-	bool canRemove() const override { return !house; }
+	bool canRemove() const override { return house.expired(); }
 
 	uint32_t getSleeper() const { return sleeperGUID; }
 
-	House* getHouse() const { return house; }
-	void setHouse(House* h) { house = h; }
+	const auto getHouse() const { return house.lock(); }
+	void setHouse(std::shared_ptr<House> house) { this->house = std::move(house); }
```

Comment:
Since it's creating a weak from a shared ptr, moving won't help here

```suggestion
	void setHouse(const std::shared_ptr<House>& house) { this->house = house; }
```